intdataplane/int_dataplane.go

```diff
@@ -203,38 +215,45 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 	dp.ifaceMonitor.Callback = dp.onIfaceStateChange
 	dp.ifaceMonitor.AddrCallback = dp.onIfaceAddrsChange
 
+	// Most iptables tables need the same options.
+	iptablesOptions := iptables.TableOptions{
+		HistoricChainPrefixes: rules.AllHistoricChainNamePrefixes,
+		InsertMode:            config.IptablesInsertMode,
+		RefreshInterval:       config.IptablesRefreshInterval,
+		PostWriteInterval:     config.IptablesPostWriteCheckInterval,
+	}
+
+	// However, the NAT tables need an extra cleanup regex.
+	iptablesNATOptions := iptablesOptions
+	iptablesNATOptions.ExtraCleanupRegexPattern = rules.HistoricInsertedNATRuleRegex
+
+	// Create the shared iptables lock.  This allows us to block other processes from
+	// manipulating iptables but it still allows us to do our work in parallel.
```

Comment:
I didn't understand the last clause on first reading this.  Perhaps "but it still allows our own goroutines to modify iptables in parallel - which is OK because they are operating on different tables"?